[APHL-545] Updated the specification of draft for multiple versions

TahaAttari opened the pull request.

Commit:
[APHL-545] Updated the specification of draft for multiple versions

Commit:
Merge branch 'master' into APHL-545-draft-operation

Commit:
[APHL-545] had main as base instead of master

sliver007 reviewed (changes requested):
$draft is not a Library-specific operation. We will need to adjust the narrative and definitions to reflect all of the resource types for which it should be supported. See the documented list of Knowledge Artifact types [here](https://simpaticois.atlassian.net/wiki/spaces/CR/pages/3096051808/Knowledge+Artifact+Resource+Types+and+Support+Priority).

Commit:
[APHL-545] update description based on feedback

Commit:
[APHL-545] phrasing

sliver007 committed:
Merge branch 'master' into APHL-545-draft-operation

sliver007 committed:
Merge branch 'master' into APHL-545-draft-operation

sliver007 committed:
Merge branch 'master' into APHL-545-draft-operation

sliver007 committed:
Changes to $draft operation to generalize

The $draft operation can be invoked on KnowledgeArtifacts, not just Library. Made some changes to indicate that - file name, resource name, url, etc. Also added the package id prefix to the operation code.

for $approve operation added the "crmi" prefix to the operation code. Also fixed the url.

sliver007 reviewed (changes requested):
The operation should apply to all knowledge artifacts, in the same way that the package operation does.

When invoking it at the type level, you need to provide enough information to identify the artifact.

https://build.fhir.org/ig/HL7/crmi-ig/operations.html

So basically, the id, url, version, identifier, and, resource parameters can all be used to identify the resource to perform the operation on.

And none of them would be required/allowed at the instance level.

Can we add the appropriate parameters to support that behavior?

Commit:
[APHL-545] updated the narrative of $draft and others so they are not Library-specific

Commit:
[APHL-545] update $draft description

sliver007 reviewed (approved):
Looks good to me!
